Inconsistent endlines in non-pretty --watch output

<!-- Please try to reproduce the issue with `typescript@next`. It may have already been fixed. -->
**TypeScript Version:**  2.9.0-dev

<!-- Search terms you tried before logging this (so others can find this issue more easily) -->
**Search Terms:** cli output pretty plain output endlines newlines

**Code**

```ts
tsc -w --pretty false
```

**Expected behavior:**

```shell
08:26:04 - Starting compilation in watch mode...

source.ts(1,9): error TS2339: Property 'yarp' does not exist on type 'Console'.

08:26:05 - Found 1 error. Watching for file changes.
```

**Actual behavior:**

```
08:25:31 - File change detected. Starting incremental compilation...


source.ts(1,9): error TS2339: Property 'yarp' does not exist on type 'Console'.
08:25:31 - Found 1 error. Watching for file changes.
```

--- a/src/compiler/watch.ts
+++ b/src/compiler/watch.ts
@@ -34,14 +34,20 @@ namespace ts {
         Diagnostics.Found_0_errors.code
     ];
 
-    function clearScreenIfNotWatchingForFileChanges(system: System, diagnostic: Diagnostic, options: CompilerOptions) {
+    /**
+     * @returns Whether the screen was cleared. 
+     */
+    function clearScreenIfNotWatchingForFileChanges(system: System, diagnostic: Diagnostic, options: CompilerOptions): boolean {
         if (system.clearScreen &&
             !options.preserveWatchOutput &&
             !options.extendedDiagnostics &&
             !options.diagnostics &&
             !contains(nonClearingMessageCodes, diagnostic.code)) {
             system.clearScreen();
+            return true;
         }
+
+        return false;
     }
 
     /** @internal */
@@ -49,12 +55,6 @@ namespace ts {
         Diagnostics.Starting_compilation_in_watch_mode.code,
         Diagnostics.File_change_detected_Starting_incremental_compilation.code,
     ];
-
-    function getPlainDiagnosticPrecedingNewLines(diagnostic: Diagnostic, newLine: string, preserveWatchOutput?: boolean): string {
-        return !preserveWatchOutput && contains(screenStartingMessageCodes, diagnostic.code)
-            ? ""
-            : newLine;
-    }
 
     function getPlainDiagnosticFollowingNewLines(diagnostic: Diagnostic, newLine: string): string {
         return contains(screenStartingMessageCodes, diagnostic.code)
@@ -74,8 +74,10 @@ namespace ts {
                 system.write(output);
             } :
             (diagnostic, newLine, options) => {
-                clearScreenIfNotWatchingForFileChanges(system, diagnostic, options);
-                let output = `${getPlainDiagnosticPrecedingNewLines(diagnostic, newLine, options.preserveWatchOutput)}${new Date().toLocaleTimeString()} - `;
+                if (!clearScreenIfNotWatchingForFileChanges(system, diagnostic, options)) {
+                    system.write(newLine);
+                }
+                let output = `${new Date().toLocaleTimeString()} - `;
                 output += `${flattenDiagnosticMessageText(diagnostic.messageText, system.newLine)}${getPlainDiagnosticFollowingNewLines(diagnostic, newLine)}`;
                 system.write(output);
             };

--- a/src/compiler/watch.ts
+++ b/src/compiler/watch.ts
@@ -35,7 +35,7 @@ namespace ts {
     ];
 
     /**
-     * @returns Whether the screen was cleared. 
+     * @returns Whether the screen was cleared.
      */
     function clearScreenIfNotWatchingForFileChanges(system: System, diagnostic: Diagnostic, options: CompilerOptions): boolean {
         if (system.clearScreen &&

--- a/src/compiler/watch.ts
+++ b/src/compiler/watch.ts
@@ -44,6 +44,24 @@ namespace ts {
         }
     }
 
+    /** @internal */
+    const screenStartingMessageCodes: number[] = [
+        Diagnostics.Starting_compilation_in_watch_mode.code,
+        Diagnostics.File_change_detected_Starting_incremental_compilation.code,
+    ];
+
+    function getPlainDiagnosticPrecedingNewLines(diagnostic: Diagnostic, newLine: string): string {
+        return contains(screenStartingMessageCodes, diagnostic.code)
+            ? ""
+            : newLine;
+    }
+
+    function getPlainDiagnosticFollowingNewLines(diagnostic: Diagnostic, newLine: string): string {
+        return contains(screenStartingMessageCodes, diagnostic.code)
+            ? newLine + newLine
+            : newLine;
+    }
+
     /**
      * Create a function that reports watch status by writing to the system and handles the formating of the diagnostic
      */
@@ -52,13 +70,13 @@ namespace ts {
             (diagnostic, newLine, options) => {
                 clearScreenIfNotWatchingForFileChanges(system, diagnostic, options);
                 let output = `[${formatColorAndReset(new Date().toLocaleTimeString(), ForegroundColorEscapeSequences.Grey)}] `;
-                output += `${flattenDiagnosticMessageText(diagnostic.messageText, system.newLine)}${newLine + newLine + newLine}`;
+                output += `${flattenDiagnosticMessageText(diagnostic.messageText, system.newLine)}${newLine + newLine}`;
                 system.write(output);
             } :
             (diagnostic, newLine, options) => {
                 clearScreenIfNotWatchingForFileChanges(system, diagnostic, options);
-                let output = new Date().toLocaleTimeString() + " - ";
-                output += `${flattenDiagnosticMessageText(diagnostic.messageText, system.newLine)}${newLine + newLine + newLine}`;
+                let output = `${getPlainDiagnosticPrecedingNewLines(diagnostic, newLine)}${new Date().toLocaleTimeString()} - `;
+                output += `${flattenDiagnosticMessageText(diagnostic.messageText, system.newLine)}${getPlainDiagnosticFollowingNewLines(diagnostic, newLine)}`;
                 system.write(output);
             };
     }

--- a/src/compiler/watch.ts
+++ b/src/compiler/watch.ts
@@ -50,8 +50,8 @@ namespace ts {
         Diagnostics.File_change_detected_Starting_incremental_compilation.code,
     ];
 
-    function getPlainDiagnosticPrecedingNewLines(diagnostic: Diagnostic, newLine: string): string {
-        return contains(screenStartingMessageCodes, diagnostic.code)
+    function getPlainDiagnosticPrecedingNewLines(diagnostic: Diagnostic, newLine: string, preserveWatchOutput?: boolean): string {
+        return !preserveWatchOutput && contains(screenStartingMessageCodes, diagnostic.code)
             ? ""
             : newLine;
     }
@@ -75,7 +75,7 @@ namespace ts {
             } :
             (diagnostic, newLine, options) => {
                 clearScreenIfNotWatchingForFileChanges(system, diagnostic, options);
-                let output = `${getPlainDiagnosticPrecedingNewLines(diagnostic, newLine)}${new Date().toLocaleTimeString()} - `;
+                let output = `${getPlainDiagnosticPrecedingNewLines(diagnostic, newLine, options.preserveWatchOutput)}${new Date().toLocaleTimeString()} - `;
                 output += `${flattenDiagnosticMessageText(diagnostic.messageText, system.newLine)}${getPlainDiagnosticFollowingNewLines(diagnostic, newLine)}`;
                 system.write(output);
             };

--- a/src/compiler/watch.ts
+++ b/src/compiler/watch.ts
@@ -74,11 +74,15 @@ namespace ts {
                 system.write(output);
             } :
             (diagnostic, newLine, options) => {
+                let output = "";
+
                 if (!clearScreenIfNotWatchingForFileChanges(system, diagnostic, options)) {
-                    system.write(newLine);
+                    output += newLine;
                 }
-                let output = `${new Date().toLocaleTimeString()} - `;
+
+                output += `${new Date().toLocaleTimeString()} - `;
                 output += `${flattenDiagnosticMessageText(diagnostic.messageText, system.newLine)}${getPlainDiagnosticFollowingNewLines(diagnostic, newLine)}`;
+
                 system.write(output);
             };
     }

--- a/src/compiler/watch.ts
+++ b/src/compiler/watch.ts
@@ -45,7 +45,7 @@ namespace ts {
     }
 
     /** @internal */
-    const screenStartingMessageCodes: number[] = [
+    export const screenStartingMessageCodes: number[] = [
         Diagnostics.Starting_compilation_in_watch_mode.code,
         Diagnostics.File_change_detected_Starting_incremental_compilation.code,
     ];

--- a/src/harness/unittests/tscWatchMode.ts
+++ b/src/harness/unittests/tscWatchMode.ts
@@ -124,7 +124,10 @@ namespace ts.tscWatch {
         }
 
         function getWatchDiagnosticWithoutDate(diagnostic: Diagnostic) {
-            return ` - ${flattenDiagnosticMessageText(diagnostic.messageText, host.newLine)}${host.newLine + host.newLine + host.newLine}`;
+            const newLines = contains(screenStartingMessageCodes, diagnostic.code)
+                ? `${host.newLine}${host.newLine}`
+                : host.newLine;
+            return ` - ${flattenDiagnosticMessageText(diagnostic.messageText, host.newLine)}${newLines}`;
         }
     }
 

